### PR TITLE
feat(elm): Add form for creating a category

### PR DESCRIPTION
**Pull request by pmrukot**
**Summary**
* Adds a form to create a new category
* Currently just supports one-word uppercase category names (should we change it?)

**Related issues**
Closes https://github.com/pmrukot/aion/issues/41

**Test plan**
Just test the form

### CODE REVIEW

File: aion/web/elm/src/Panel/Api.elm [hunk 1]
```diff
@@ -6,12 +6,15 @@ import General.Utils exposing (getSubjectIdByName)
...
 import Json.Encode as Encode
-import Panel.Models exposing (QuestionForm)
+import Panel.Models exposing (CategoryForm, QuestionForm)
 import Room.Models exposing (RoomsData)
 
 
+-- create question section
+
+
 createQuestionUrl : String
```
Comment (mrapacz) [inline aion/web/elm/src/Panel/Api.elm]: I think you can move the Urls to Constants.elm

File: aion/web/elm/src/Panel/Notifications.elm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module Panel.Notifications exposing (..)
+
+import General.Models exposing (Model)
+import General.Notifications exposing (addToast)
+import Msgs exposing (Msg(..))
+import Toasty.Defaults
+
+
+categoryFormValidationErrorToast : ( Model, Cmd Msg ) -> ( Model, Cmd Msg )
```
Comment (mrapacz) [inline aion/web/elm/src/Panel/Notifications.elm]: Here I have a similar comment as to the previous PR. If we've put three functions mirroring three types of notifications in src/Notifications.elm, then this file could only consist of three strings + import Notifications. I think that would make it simpler, but, as I said, we can delay it until we'll be dealing with global elm refactor.

File: aion/web/elm/src/Panel/Validators.elm [hunk 3]
```diff
@@ -31,6 +31,17 @@ subjectValidations =
     [ Forms.validateExistence ]
 
 
+categoryNameValidations : List Forms.FieldValidator
+categoryNameValidations =
+    let
+        categoryNameRegex =
+            "\\b([A-Z][a-z]+)\\b"
```
Comment (mrapacz) [inline aion/web/elm/src/Panel/Validators.elm]:
```
^\s*(.+?)\s+$
```
You could do this for multi word categories. Did you check how your regex behaves when dealing with Unicode characters?

File: aion/web/elm/src/Update.elm [hunk 4]
```diff
@@ -50,6 +52,24 @@ update msg model =
                 _ ->
                     model ! []
 
+        OnCategoryCreated response ->
+            case response of
+                RemoteData.Success responseData ->
```
Comment (mrapacz) [inline aion/web/elm/src/Update.elm]: This just cries for a proper setter method, cannot wait for #56

**Review by mrapacz — CHANGES_REQUESTED** [review]
I think the regex part could work better here, especially the Unicode. The rest is highly analogous to #61, which means it's correct, but cannot wait for #56

**Comment by pmrukot** [comment]
Updated branch and adjusted code to your suggestions. Lets leave regex discussion for another day..

**Review by mrapacz — APPROVED** [review]

**Comment by mrapacz** [comment]
giphy 1

**Pull request merged**